safe_cli/operators/hw_accounts/trezor_manager.py

```diff
@@ -0,0 +1,51 @@
+from functools import lru_cache
+
+from eth_typing import ChecksumAddress
+from trezorlib import tools
+from trezorlib.client import TrezorClient, get_default_client
+from trezorlib.ethereum import get_address, sign_typed_data_hash
+from trezorlib.ui import ClickUI
+
+from safe_cli.operators.hw_accounts.hw_account import HwAccount
+from safe_cli.operators.hw_accounts.trezor_exceptions import (
+    raise_trezor_exception_as_hw_account_exception,
+)
+
+
+@lru_cache(maxsize=None)
+@raise_trezor_exception_as_hw_account_exception
+def get_trezor_client() -> TrezorClient:
```

Comment:
Why not having `TrezorManager` as the singleton class (as you do with Ledger) instead of doing that with `TrezorClient`?

Reply:
LedgerWallet is not a singleton, the ledgereth library is caching internally the dongle.

Reply:
Understood